packages/react-app/pages/admin/batches.jsx

```diff
@@ -115,15 +117,38 @@ export default function Batches({ serverUrl, userRole, mainnetProvider }) {
     setGraduatesCount(graduatesCounts);
   }, [batches, serverUrl]);
 
+  const fetchParticipantsCount = useCallback(async () => {
+    const fetchedBatchParticipants = await axios.get(serverUrl + serverPathBatchParticipants);
+    const participantsCounts = {};
+
+    batches.forEach(
+      batch => {
+        const batchName = batch.batchName;
+        participantsCounts[batchName] = 0;
+
+        fetchedBatchParticipants.data.forEach(builder => {
+          if (builder.batch && builder.batch.number === String(batchName)) {
+            participantsCounts[batchName]++;
+          }
+        });
+      },
+      [batches],
+    );
+
+    setParticipantsCount(participantsCounts);
+  }, [batches, serverUrl]);
```

Comment:
Damn, this nested loop is ugly haha

We are also making 3 requests to the server to render this page, which doesn't make sense. 

Let's update the backend as needed so we don't need to do this. We could:

- Extend the /batches endpoint so it comes back with the count calculations
or
- Create a new `/batches/stats` route that includes this data

So we'll only need to do 1 request. Also will be helpful here https://github.com/BuidlGuidl/buidlguidl.com/pull/54#discussion_r1830737984